### PR TITLE
Fix fog mock update instance while attach volume

### PR DESCRIPTION
**Pull request by alex-lairan**

### CODE REVIEW

**Comment by alex-lairan** [comment]
Hi, this fix https://github.com/fog/fog-aws/issues/236 problem.

In hope you like it :)

**cc**: @plribeiro3000 @shaiguitar @geemus

**Comment by alex-lairan** [comment]
The test code

test.rb
`require 'fog'
require 'test/unit'

#
class MyTest < Test::Unit::TestCase
  # rubocop:disable Metrics/AbcSize
  # rubocop:disable Metrics/MethodLength
  CREDS = {
    provider: 'AWS',
    region: 'eu-west-1',
    aws_access_key_id: 'foo',
    aws_secret_access_key: 'bar'
  }.freeze

  VM_DATA = {
    name: 'fogtest',
    image_name: 'ami-8cea48ff',
    machine_type: 't1.micro',
    zone_name: 'eu-west-1'
  }.freeze

  def test_volume
    Fog.mock!

    fogc = Fog::Compute.new(CREDS)

    vm = fogc.servers.create(VM_DATA)
    vm.wait_for { ready? }

    # creating and attaching volume
    vol = fogc.volumes.create(availability_zone: 'eu-west-1', size: 1)
    vol.wait_for { ready? }
    fogc.attach_volume(vm.id, vol.id, '/dev/sdi')
    vol.wait_for { state == 'in-use' }

    # volume exists and is attached
    vol.reload
    assert_not_nil vol.server_id
    assert_equal 'in-use', vol.state

    # vm should have this volume listed
    vm.reload
    assert_not_empty vm.block_device_mapping
  end
  # rubocop:enable Metrics/AbcSize
  # rubocop:enable Metrics/MethodLength
end
`

Gemfile
`source 'https://rubygems.org'

gem 'fog'
gem 'test-unit'

gem 'fog-aws', path: '~/Documents/fog-aws'
`

File: lib/fog/aws/requests/compute/attach_volume.rb [hunk 1]
```diff
@@ -53,6 +53,8 @@ def attach_volume(instance_id, volume_id, device)
                 'status'      => 'attaching',
                 'volumeId'    => volume_id
               }
+              # Insert new mapping information into the instance
+              instance['blockDeviceMapping'] << data
```
Comment (lanej) [inline lib/fog/aws/requests/compute/attach_volume.rb]: I believe `data` needs to be transformed into [the block device mapping type](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_BlockDeviceMapping.html)

**Comment by alex-lairan** [comment]
So I need to transform 
```{"attachTime"=>2016-04-28 10:32:52 +0200,
 "device"=>"/dev/sdi",
 "instanceId"=>"i-c8fab1a8",
 "status"=>"attaching",
 "volumeId"=>"vol-1372a876"}``` into 
```{"deviceName"=>"/dev/sdg",
  "volumeId"=>"vol-91fc2c63",
  "status"=>"attached",
  "attachTime"=>2016-03-17 17:11:30 UTC,
  "deleteOnTermination"=>false}``` right? :)

(The both values are different, juste compare format :p)

**Comment by alex-lairan** [comment]
When I puts I get [{"deviceName"=>"/dev/sdi",
  "volumeId"=>"vol-4eb43027",
  "status"=>"attaching",
  "attachTime"=>2016-04-28 09:00:18 UTC,
  "deleteOnTermination"=>"false"}]

Where I can found **deleteOnTermination** information? I've let's a puts if you want test :)

**Comment by lanej** [comment]
@alex-lairan you have to dig int

**Comment by lanej** [comment]
@alex-lairan can you add a test for this ?

**Comment by lanej** [comment]
Stale. Closing.

**Pull request closed without merging**